test-runner/src/main.rs

```diff
@@ -6,46 +6,90 @@
     trivial_casts,
     unused_qualifications
 )]
-/// I am just a simple worker staying busy with one WebAssembly program at a time.
 use clap::Parser;
+use engine::extensions::load_extensions;
 use owo_colors::OwoColorize;
-use std::collections::HashMap;
+
+use std::fs;
+use std::fs::File;
 use std::io::{stdin, Read};
-use std::path::Path;
-use std::{ffi::OsStr, fs};
+use std::path::{Path, PathBuf};
+use std::process::exit;
+use std::str::FromStr;
+use utils::structs::{Manifest, Permission};
 
 use engine::ServalEngine;
 
 /// Note: The CLI is just here for simple testing purpose.
 /// The real worker will pick up executables and inputs from an API endpoint.
 #[derive(Parser, Debug)]
 struct CLIArgs {
-    /// Path to the WASM executable to run
+    /// Path to the WASM executable or Serval manifest to run
     // TODO: Check for the WASM binary magic bytes [1] or even evaluate file grammar [2].
     // [1]: Example: https://developer.mozilla.org/en-US/docs/WebAssembly/Understanding_the_text_format#the_simplest_module
     // [2]: Specification: https://webassembly.github.io/spec/core/index.html
-    exec_path: String,
+    exec_path: PathBuf,
     /// Optional path to a file containing input for the executable
     // Naive initial approach: We don't check file content and assume the executable knows what to do with it.
     // TODO: How would we validate that an input file is "correct" without running the job and seeing if it fails? TBD.
-    input_path: Option<String>,
+    #[clap(long)]
+    input_path: Option<PathBuf>,
+    /// Optional path to a directory full of Serval extensions
+    #[clap(long)]
+    extensions_path: Option<PathBuf>,
+    #[clap(long)]
+    permissions: Option<String>,
 }
 
 fn main() -> anyhow::Result<()> {
     let args = CLIArgs::parse();
 
-    let exec_path = Path::new(&args.exec_path);
-    let extension = exec_path.extension().and_then(OsStr::to_str);
+    let manifest = match args
+        .exec_path
+        .extension()
+        .unwrap_or_default()
+        .to_string_lossy()
+        .as_ref()
+    {
+        "toml" => {
+            let manifest = Manifest::from_file(&args.exec_path)?;
+
+            eprintln!(
+                "{} {}",
+                "manifest:".blue().bold(),
+                toml::to_string(&manifest)
+                    .unwrap()
+                    .replace('\n', "\n          ")
```

Comment:
Man, I stared at this line for a good while until it clicked. Not a critique, I definitely need this in another place 😅

Reply:
In JavaScript Land I would've written something like `"manifest: ".replace(/./g, " ")` but it would be so much more verbose in  Rust Land...